test/test_read.py

```diff
@@ -2,17 +2,36 @@
 from pathlib import Path
 from mdframe.reader import run, Config
 import json
+import toml
+
 class TestReader(unittest.TestCase):
 
     def test_metadata_print(self):
         data_dir_path = Path(__file__).parent / "data_dir"
+        schema_loc = Path('../src/mdframe/schema.json')
         config = Config(
             data_dir_path=data_dir_path, 
             metadata_file_extension="toml", 
-            schema_url='../src/mdframe/schema.json'
+            schema_loc=schema_loc
         )
         df = run(config)
         print(df)
 
+    def test_metadata_invalid(self):
+        data_dir_path = Path(__file__).parent / "invalid_data_dir"
+        schema_loc = Path('../src/mdframe/schema.json')
+        config = Config(
+            data_dir_path=data_dir_path, 
+            metadata_file_extension="toml", 
+            schema_loc=schema_loc
+        )
+
+        # expecting this to fail
+        try:
```

Comment:
You should use an `xfail` here. Check out `self.assertRaises`, for example:

```python
def test_split(self):
        s = 'hello world'
        self.assertEqual(s.split(), ['hello', 'world'])
        # check that s.split fails when the separator is not a string
        with self.assertRaises(TypeError):
            s.split(2)
```

Reply:
taken care of!